Solutions/GitLab/Analytic Rules/GitLab_BruteForce.yaml

```diff
@@ -0,0 +1,51 @@
+id: 2238d13a-cf05-4973-a83f-d12a25dbb153
+name: GitLab - Brute-force Attempts
+description: |
+  'This query relies on GitLab Application Logs to get failed logins to highlight brute-force attempts from different IP addresses in a short space of time.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: Syslog
+    dataTypes: 
+      - Syslog
+queryFrequency: 1h
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+relevantTechniques:
+  - T1110
+query: |
+  let LearningPeriod = 7d; 
+  let BinTime = 1h; 
+  let RunTime = 1h; 
+  let StartTime = 1h; 
+  let NumberOfStds = 3; 
+  let MinThreshold = 10.0; 
+  let EndRunTime = StartTime - RunTime; 
+  let EndLearningTime = StartTime + LearningPeriod;
+  let GitLabFailedLogins = (GitLabApp
+  | where Message contains "Failed Login"
```

Comment:
I think I understand why you are using contains here versus has, but I think the best solution would be to use the parser you built to parse the Message field so that using contains here is not needed.